### DISCUSSION TITLE
Mouse emulation appears to not be compiling since ZMK main and/ urob/zmk update

### DISCUSSION THREAD

**Discussion by fsqonbdk** [manna-harbour/miryoku#265] [Q&A] [answered]
I was able to compile successfully and have full working mouse support on a Corne with nice!nano v2s as recently as 10/17/2023 (date of most recent working compile).

 Running build-example-mousekeyspr.yml workflow now generates the following errors:

`  From https://github.com/urob/zmk
   * branch              mouse-3.2  -> FETCH_HEAD
   * [new branch]        mouse-3.2  -> urob-zmk/mouse-3.2
  CONFLICT (add/add): Merge conflict in docs/docs/behaviors/mouse-emulation.md
  Auto-merging docs/docs/behaviors/mouse-emulation.md
  Auto-merging app/src/hog.c
  CONFLICT (content): Merge conflict in app/src/hog.c
  Auto-merging app/src/hid.c
  CONFLICT (content): Merge conflict in app/src/hid.c
  CONFLICT (add/add): Merge conflict in app/src/events/mouse_button_state_changed.c
  Auto-merging app/src/events/mouse_button_state_changed.c
  Auto-merging app/src/endpoints.c
  CONFLICT (content): Merge conflict in app/src/endpoints.c
  CONFLICT (add/add): Merge conflict in app/src/behaviors/behavior_mouse_key_press.c
  Auto-merging app/src/behaviors/behavior_mouse_key_press.c
  CONFLICT (add/add): Merge conflict in app/include/zmk/mouse.h
  Auto-merging app/include/zmk/mouse.h
  Auto-merging app/include/zmk/hog.h
  CONFLICT (content): Merge conflict in app/include/zmk/hog.h
  Auto-merging app/include/zmk/hid.h
  CONFLICT (content): Merge conflict in app/include/zmk/hid.h
  CONFLICT (add/add): Merge conflict in app/include/zmk/events/mouse_button_state_changed.h
  Auto-merging app/include/zmk/events/mouse_button_state_changed.h
  Auto-merging app/include/zmk/endpoints.h
  CONFLICT (content): Merge conflict in app/include/zmk/endpoints.h
  CONFLICT (add/add): Merge conflict in app/include/dt-bindings/zmk/mouse.h
  Auto-merging app/include/dt-bindings/zmk/mouse.h
  CONFLICT (add/add): Merge conflict in app/dts/behaviors/mouse_key_press.dtsi
  Auto-merging app/dts/behaviors/mouse_key_press.dtsi
  Auto-merging app/dts/behaviors.dtsi
  CONFLICT (content): Merge conflict in app/dts/behaviors.dtsi
  Auto-merging app/Kconfig
  CONFLICT (content): Merge conflict in app/Kconfig
  Auto-merging app/CMakeLists.txt
  CONFLICT (content): Merge conflict in app/CMakeLists.txt
  Automatic merge failed; fix conflicts and then commit the result.
  Error: Process completed with exit code 1.`

Also attempted removing urob/zmk merge reference and got the following errors:

`  -- west build: generating a build system
  Loading Zephyr default modules (Zephyr base).
  -- Application: /__w/miryoku_zmk/miryoku_zmk/zmk/app
  -- Adding /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/corne
  -- Adding /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/nice_view_adapter
  -- Adding /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/nice_view
  -- ZMK Config directory: /__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config
  -- ZMK Config Kconfig: /__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config/corne_left.conf
  -- Using keymap file: /__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config/corne.keymap
  -- Using keymap file: /__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config/corne.keymap
  -- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
  -- Cache files will be written to: /__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/.cache
  -- Zephyr version: 3.2.0 (/__w/miryoku_zmk/miryoku_zmk/zmk/zephyr)
  -- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
  -- Board: nice_nano_v2
  -- Shield(s): corne_left nice_view_adapter nice_view
  -- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
  -- Found host-tools: zephyr 0.15.2 (/opt/zephyr-sdk-0.15.2)
  -- Found toolchain: zephyr 0.15.2 (/opt/zephyr-sdk-0.15.2)
  -- Found Dtc: /opt/zephyr-sdk-0.15.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
  -- Found BOARD.dts: /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/arm/nice_nano/nice_nano_v2.dts
  -- Found devicetree overlay: /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/corne/corne_left.overlay
  -- Found devicetree overlay: /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/corne/boards/nice_nano_v2.overlay
  -- Found devicetree overlay: /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/nice_view_adapter/nice_view_adapter.overlay
  -- Found devicetree overlay: /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/nice_view_adapter/boards/nice_nano_v2.overlay
  -- Found devicetree overlay: /__w/miryoku_zmk/miryoku_zmk/zmk/app/boards/shields/nice_view/nice_view.overlay
  -- Found devicetree overlay: /__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config/corne.keymap
  devicetree error: /__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config/../miryoku/miryoku.dtsi:21 (column 6961): parse error: expected number or parenthesized expression
  -- In: /__w/miryoku_zmk/miryoku_zmk/zmk/app/build/zephyr, command: /usr/bin/python3.8;/__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/scripts/dts/gen_defines.py;--dts;/__w/miryoku_zmk/miryoku_zmk/zmk/app/build/zephyr/zephyr.dts.pre;--dtc-flags;'';--bindings-dirs;/__w/miryoku_zmk/miryoku_zmk/zmk/app/dts/bindings;/__w/miryoku_zmk/miryoku_zmk/zmk/app/module/dts/bindings;/__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/dts/bindings;--header-out;/__w/miryoku_zmk/miryoku_zmk/zmk/app/build/zephyr/include/generated/devicetree_generated.h.new;--dts-out;/__w/miryoku_zmk/miryoku_zmk/zmk/app/build/zephyr/zephyr.dts.new;--edt-pickle-out;/__w/miryoku_zmk/miryoku_zmk/zmk/app/build/zephyr/edt.pickle;--vendor-prefixes;/__w/miryoku_zmk/miryoku_zmk/zmk/app/dts/bindings/vendor-prefixes.txt;--vendor-prefixes;/__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/dts/bindings/vendor-prefixes.txt
  CMake Error at /__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/cmake/modules/dts.cmake:230 (message):
    gen_defines.py failed with return code: 1
  Call Stack (most recent call first):
    /__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/cmake/modules/zephyr_default.cmake:108 (include)
    /__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
    /__w/miryoku_zmk/miryoku_zmk/zmk/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
    CMakeLists.txt:15 (find_package)
  
  
  -- Configuring incomplete, errors occurred!
  FATAL ERROR: command exited with status 1: /usr/local/bin/cmake -DWEST_PYTHON=/usr/bin/python3 -B/__w/miryoku_zmk/miryoku_zmk/zmk/app/build -GNinja -DBOARD=nice_nano_v2 '-DSHIELD=corne_left nice_view_adapter nice_view' -DZMK_CONFIG=/__w/miryoku_zmk/miryoku_zmk/miryoku_zmk/config -S/__w/miryoku_zmk/miryoku_zmk/zmk/app
  Error: Build failed with exit code 1.
  Error: Process completed with exit code 1.`

**Comment by manna-harbour** [comment, marked as answer]
See https://github.com/manna-harbour/miryoku/discussions/208#discussioncomment-7641348.
